1.18.2 fails to build for 32bit Windows

clang:

```
  D:\M\msys64\clang32\bin\clang++.exe -DENABLE_MULTITHREADING_SUPPORT=1 -DENABLE_PARALLEL_TILE_DECODING=1 -DHAVE_AOM_DECODER=1 -DHAVE_AOM_ENCODER=1 -DHAVE_BROTLI=1 -DHAVE_DAV1D=1 -DHAVE_JPEG_DECODER=1 -DHAVE_JPEG_ENCODER=1 -DHAVE_KVAZAAR=1 -DHAVE_KVAZAAR_ENABLE_LOGGING=1 -DHAVE_LIBDE265=1 -DHAVE_LIBSHARPYUV=1 -DHAVE_OPENJPEG_DECODER=1 -DHAVE_OPENJPEG_ENCODER=1 -DHAVE_RAV1E=1 -DHAVE_UNISTD_H -DHAVE_VISIBILITY -DHAVE_X265=1 -DHAVE_ZLIB=1 -DLIBHEIF_EXPORTS -Dheif_EXPORTS -IC:/_/B/src/build-CLANG32 -IC:/_/B/src/libheif-1.18.2/libheif -IC:/_/B/src/libheif-1.18.2/libheif/api -IC:/_/B/src/libheif-1.18.2/include/libheif -IC:/_/B/src/libheif-1.18.2/include -ID:/M/msys64/clang32/include/rav1e -ID:/M/msys64/clang32/include/webp -isystem D:/M/msys64/clang32/include/openjpeg-2.5 -march=pentium4 -mtune=generic -O2 -pipe -Wp,-D_FORTIFY_SOURCE=2 -fstack-protector-strong -O3 -DNDEBUG -std=c++11 -fvisibility=hidden -fvisibility-inlines-hidden   -Wall -Werror -Wsign-compare -Wconversion -Wno-sign-conversion -Wno-error=conversion -Wno-error=unused-parameter -Wno-error=deprecated-declarations -Wno-error=tautological-compare -Wno-error=tautological-constant-out-of-range-compare -Wno-error=potentially-evaluated-expression -MD -MT libheif/CMakeFiles/heif.dir/plugins/decoder_openjpeg.cc.obj -MF libheif\CMakeFiles\heif.dir\plugins\decoder_openjpeg.cc.obj.d -o libheif/CMakeFiles/heif.dir/plugins/decoder_openjpeg.cc.obj -c C:/_/B/src/libheif-1.18.2/libheif/plugins/decoder_openjpeg.cc
  C:/_/B/src/libheif-1.18.2/libheif/plugins/decoder_openjpeg.cc:259:56: error: no matching constructor for initialization of 'std::unique_ptr<opj_codec_t, void (*)(opj_codec_t *)>' (aka 'unique_ptr<void *, void (*)(void **)>')
    259 |   std::unique_ptr<opj_codec_t, void (*)(opj_codec_t*)> l_codec(opj_create_decompress(OPJ_CODEC_J2K),
        |                                                        ^       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    260 |                                                                opj_destroy_codec);
```

gcc:

```
  D:\M\msys64\mingw32\bin\g++.exe -DENABLE_MULTITHREADING_SUPPORT=1 -DENABLE_PARALLEL_TILE_DECODING=1 -DHAVE_AOM_DECODER=1 -DHAVE_AOM_ENCODER=1 -DHAVE_BROTLI=1 -DHAVE_DAV1D=1 -DHAVE_JPEG_DECODER=1 -DHAVE_JPEG_ENCODER=1 -DHAVE_KVAZAAR=1 -DHAVE_KVAZAAR_ENABLE_LOGGING=1 -DHAVE_LIBDE265=1 -DHAVE_LIBSHARPYUV=1 -DHAVE_OPENJPEG_DECODER=1 -DHAVE_OPENJPEG_ENCODER=1 -DHAVE_RAV1E=1 -DHAVE_UNISTD_H -DHAVE_VISIBILITY -DHAVE_X265=1 -DHAVE_ZLIB=1 -DLIBHEIF_EXPORTS -Dheif_EXPORTS -IC:/_/B/src/build-MINGW32 -IC:/_/B/src/libheif-1.18.2/libheif -IC:/_/B/src/libheif-1.18.2/libheif/api -IC:/_/B/src/libheif-1.18.2/include/libheif -IC:/_/B/src/libheif-1.18.2/include -ID:/M/msys64/mingw32/include/rav1e -ID:/M/msys64/mingw32/include/webp -isystem D:/M/msys64/mingw32/include/openjpeg-2.5 -march=pentium4 -mtune=generic -O2 -pipe -Wp,-D_FORTIFY_SOURCE=2 -fstack-protector-strong -O3 -DNDEBUG -std=c++11 -fvisibility=hidden -fno-keep-inline-dllexport   -Wall -Werror -Wsign-compare -Wconversion -Wno-sign-conversion -Wno-error=conversion -Wno-error=unused-parameter -Wno-error=deprecated-declarations -MD -MT libheif/CMakeFiles/heif.dir/plugins/decoder_openjpeg.cc.obj -MF libheif\CMakeFiles\heif.dir\plugins\decoder_openjpeg.cc.obj.d -o libheif/CMakeFiles/heif.dir/plugins/decoder_openjpeg.cc.obj -c C:/_/B/src/libheif-1.18.2/libheif/plugins/decoder_openjpeg.cc
  C:/_/B/src/libheif-1.18.2/libheif/plugins/decoder_openjpeg.cc: In function 'heif_error openjpeg_decode_image(void*, heif_image**)':
  C:/_/B/src/libheif-1.18.2/libheif/plugins/decoder_openjpeg.cc:260:81: error: no matching function for call to 'std::unique_ptr<void*, void (*)(void**)>::unique_ptr(void**, void (__attribute__((stdcall)) &)(void**))'
    260 |                                                                opj_destroy_codec);
```

--- a/libheif/plugins/decoder_openjpeg.cc
+++ b/libheif/plugins/decoder_openjpeg.cc
@@ -256,7 +256,7 @@ struct heif_error openjpeg_decode_image(void* decoder_raw, struct heif_image** o
 
   OPJ_BOOL success;
   opj_dparameters_t decompression_parameters;
-  std::unique_ptr<opj_codec_t, void (*)(opj_codec_t*)> l_codec(opj_create_decompress(OPJ_CODEC_J2K),
+  std::unique_ptr<opj_codec_t, void (OPJ_CALLCONV *)(opj_codec_t*)> l_codec(opj_create_decompress(OPJ_CODEC_J2K),
                                                                opj_destroy_codec);
 
   // Initialize Decoder
@@ -271,7 +271,7 @@ struct heif_error openjpeg_decode_image(void* decoder_raw, struct heif_image** o
   // Create Input Stream
 
   OPJ_BOOL is_read_stream = true;
-  std::unique_ptr<opj_stream_t, void (*)(opj_stream_t*)> stream(opj_stream_create_default_memory_stream(decoder, is_read_stream),
+  std::unique_ptr<opj_stream_t, void (OPJ_CALLCONV *)(opj_stream_t*)> stream(opj_stream_create_default_memory_stream(decoder, is_read_stream),
                                                                 opj_stream_destroy);
 
 
@@ -283,7 +283,7 @@ struct heif_error openjpeg_decode_image(void* decoder_raw, struct heif_image** o
     return err;
   }
 
-  std::unique_ptr<opj_image_t, void (*)(opj_image_t*)> image(image_ptr, opj_image_destroy);
+  std::unique_ptr<opj_image_t, void (OPJ_CALLCONV *)(opj_image_t*)> image(image_ptr, opj_image_destroy);
 
   if (image->numcomps != 3 && image->numcomps != 1) {
     //TODO - Handle other numbers of components

